code/modules/mob/living/carbon/xenomorph/update_icons.dm

```diff
@@ -39,6 +39,15 @@
 			Q.queen_standing_icon = icon_xeno
 			Q.queen_ovipositor_icon = get_icon_from_source(CONFIG_GET(string/alien_queen_ovipositor))
 
+	var/mutation_caste_state = "[mutation_type] [caste.caste_type]"
+	if(!walking_state_cache[mutation_caste_state])
+		var/cache_walking_state = FALSE
+		for(var/state in icon_states(icon))
+			if(findtext(state, "Walking"))
```

Comment:
whew that part's pretty dubious

what's the point/problem at stake? can't we just define on castes which has walks and which not?
i think ideally we'd just want all of them to have placeholder icons

Reply:
my main concern is for stuff like strains that may have running and not walking, using this automated check we don't have to worry about going through each one and setting the correct one, also future-proofs it

Reply:
Ugh, in all honesty this looks more like cornercutting to me... There's no real reason we'd not just have them all defined like the rest -- and even then it could just be a flag on xenos modified by apply_mutator

if you don't mind carlarc's son posting this on discord, i guess we can merge it

Reply:
truly we will be memed on for this but that's okay